atomic_reactor/plugins/pre_add_image_content_manifest.py

```diff
@@ -0,0 +1,186 @@
+                session.verify = True
+                return session
+            session = make_session()
+            response = session.get(self.icm_url)
+            response.raise_for_status()
+            self._icm = response.json()  # Returns dict
+            # Validate; `json.dumps()` converts `icm` to str. Confusingly, `read_yaml`
+            #     *will* validate JSON
+            read_yaml(json.dumps(self._icm), 'schemas/content_manifest.json')
+        return self._icm
+
+    def _populate_content_sets(self):
+        """
+        Get the list of the current platform's content_sets from
+        'content_sets.yml' in dist-git, and set `self.content_sets` to same.
+        """
+        current_platform = self.workflow.user_params['platform']
+        workdir = self.workflow.builder.df_dir
+        fpath = os.path.join(workdir, REPO_CONTENT_SETS_CONFIG)
+        if os.path.exists(fpath):
+            content_sets_yml = read_yaml_from_file_path(fpath, 'schemas/content_sets.json') or {}
+            if current_platform in content_sets_yml:
+                self.content_sets = content_sets_yml[current_platform]
+        self.log.debug('Output content_sets: %s', self.content_sets)
+
+    def _update_icm_data(self):
+        # Inject the content_sets data into the ICM JSON object
+        self.icm['content_sets'] = self.content_sets
+
+        # Inject the current image layer index number into the ICM JSON object metadata
+        self.icm['metadata']['image_layer_index'] = self.layer_index
+
+        # Convert dict -> str
+        icm_json = json.dumps(self.icm, indent=4)
+
+        # Validate the ICM with the ICM JSON Schema
+        read_yaml(icm_json, 'schemas/content_manifest.json')
+
+        self.log.debug('Output ICM: %s', icm_json)
+
+    def _write_json_file(self):
+        out_file_path = os.path.join(self.workflow.builder.df_dir,
+                                     self.icm_file_name)
+        if os.path.exists(out_file_path):
+            raise RuntimeError('File {} already exists in repo'.format(out_file_path))
+        with open(out_file_path, 'w') as outfile:
+            json.dump(self.icm, outfile, indent=4)
+        self.log.debug('ICM JSON saved to: %s', out_file_path)
+
+    def _add_to_dockerfile(self):
+        """
+        Put an ADD instruction into the Dockerfile (to include the ICM file
+        into the container image to be built)
+        """
+        content = 'ADD {0} {1}'.format(self.icm_file_name, self.content_manifests_dir)
+        lines = self.dfp.lines
+
+        # Put it before last instruction
+        lines.insert(-1, content + '\n')
+        self.dfp.lines = lines
+
+    def run(self):
+        """
+        run the plugin
+        """
+        self._populate_content_sets()
+        self._update_icm_data()
+        self._write_json_file()
+        self._add_to_dockerfile()
+        self.log.info("added %s", self.content_manifests_dir)
```

Comment:
Should this also include the file name added?

Reply:
ack